libraries/api/discussion_helper.cpp

```diff
@@ -183,8 +183,69 @@ namespace golos { namespace api {
 
             if (d.allow_curation_rewards) {
                 if (d.total_vote_weight > 0) {
+
+                    uint128_t votes_in_auction_window_weight;
+                    uint128_t votes_after_auction_window_weight;
+
+                    share_type votes_in_auction_window_reward;
+                    share_type votes_after_auction_window_reward;
+                    share_type auction_window_reward;
+
+                    auto auw_time = d.created + d.auction_window_size;
+
+                    if (db.has_hardfork(STEEMIT_HARDFORK_0_19__898) && 
+                        d.auction_window_reward_destination == protocol::to_curators
+                    ) {
+                        // separate votes
+                        const auto &cvlupdidx = db.get_index<comment_vote_index>().indices().get<by_vote_last_update>();
+                        auto itr = cvlupdidx.lower_bound(boost::make_tuple(0, d.created));
+                        auto itr_after_auw = cvlupdidx.lower_bound( // auw -- auctcion window 
+                            boost::make_tuple(0, d.created + d.auction_window_size)
+                        );
+
+                        while (itr != itr_after_auw && itr->comment == d.id) {
```

Comment:
You shouldn't do such iterations on runtime - it's too heavy and will generate forks.

Reply:
Do it in the vote evaluator.

Reply:
Yes it's really heavy and extra loop. Added one more field in `comment_object` and added needed calculations to `vote_evaluator`